rpm/generic/zfs.spec.in

```diff
@@ -34,31 +34,14 @@
 %endif
 
 %bcond_with    debug
-%bcond_with    systemd
+%bcond_without systemd
 
-# Generic enable switch for systemd
-%if %{with systemd}
+# Enable systemd when _unitdir is defined or with generic enable switch.
+%if %{with systemd} && %{defined _unitdir}
 %define _systemd 1
```

Comment:
You can actually take `_systemd 1` out entirely and replace it with nothing. The `%bcond` definitions actually define a variable for you. In this case, it's `with_systemd`. Nothing extra required. See: https://github.com/rpm-software-management/rpm/blob/master/macros.in#L104-L140